apps/shipments/serializers/tags.py

```diff
@@ -0,0 +1,52 @@
+"""
+Copyright 2020 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from django.conf import settings
+from rest_framework.validators import UniqueTogetherValidator
+from rest_framework_json_api import serializers
+
+from apps.permissions import get_owner_id
+from ..models import ShipmentTag
+
+
+class ShipmentTagSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = ShipmentTag
+        fields = '__all__'
+
+
+class ShipmentTagCreateSerializer(ShipmentTagSerializer):
+    shipment_id = serializers.CharField(required=False)
+
+    def __init__(self, *args, **kwargs):
```

Comment:
No need to override __init__ here - I believe we can just override validate_shipment_id and validate_owner_id, and add a check to populate them from the context if they don't exist yet/PROFILES_ENABLED.

Reply:
This one is tricky. When trying to add those two methods instead, I got a `400` error due to either of those values missing for the `initial_data`. Since in order to automatically run the UniqueTogetherValidator validation on those fields, they need to be declared in Meta.fields.

Reply:
Ah, I see. I thought validate_{field} ran before the other validations, and we could just populate them there. In that case this is fine, thanks for the explanation!